src/Service.Tests/TestHelper.cs

```diff
@@ -237,5 +240,79 @@ public static void AddMissingEntitiesToConfig(RuntimeConfig config, string entit
             },
             ""entities"": {}" +
           "}";
+
+        /// <summary>
+        /// A valid REST API request body with correct parameter types for all the fields.
+        /// </summary>
+        public const string REQUEST_BODY_WITH_CORRECT_PARAM_TYPES = @"
+                    {
+                        ""title"": ""New book"",
+                        ""publisher_id"": 1234
+                    }
+                ";
+
+        /// <summary>
+        /// An invalid REST API request body with incorrect parameter type for publisher_id field.
+        /// </summary>
+        public const string REQUEST_BODY_WITH_INCORRECT_PARAM_TYPES = @"
+                    {
+                        ""title"": ""New book"",
+                        ""publisher_id"": ""one""
+                    }
+                ";
+
+        /// <summary>
+        /// Utility method that reads the config file for a given database type and constructs a
+        /// new config file with changes just in the host mode section.
+        /// </summary>
+        /// <param name="configFileName">Name of the new config file to be constructed</param>
+        /// <param name="hostModeType">HostMode for the engine</param>
+        /// <param name="databaseType">Database type</param>
+        public static void ConstructNewConfigWithSpecifiedHostMode(string configFileName, HostModeType hostModeType, string databaseType)
+        {
+            RuntimeConfigProvider configProvider = TestHelper.GetRuntimeConfigProvider(databaseType);
+            RuntimeConfig config = configProvider.GetRuntimeConfiguration();
+            HostGlobalSettings customHostGlobalSettings = config.HostGlobalSettings with { Mode = hostModeType };
+            JsonElement serializedCustomHostGlobalSettings =
+                JsonSerializer.SerializeToElement(customHostGlobalSettings, RuntimeConfig.SerializerOptions);
+            Dictionary<GlobalSettingsType, object> customRuntimeSettings = new(config.RuntimeSettings);
+            customRuntimeSettings.Remove(GlobalSettingsType.Host);
+            customRuntimeSettings.Add(GlobalSettingsType.Host, serializedCustomHostGlobalSettings);
+            RuntimeConfig configWithCustomHostMode =
+                config with { RuntimeSettings = customRuntimeSettings };
+            File.WriteAllText(
+                configFileName,
+                JsonSerializer.Serialize(configWithCustomHostMode, RuntimeConfig.SerializerOptions));
+
+        }
+
+        /// <summary>
+        /// Utility method that fetches the HTTP method as HttpMethod type
+        /// from the given string type.
+        /// </summary>
+        /// <param name="httpMethod"></param>
+        /// <returns></returns>
+        /// <exception cref="DataApiBuilderException"></exception>
+        public static HttpMethod GetHttpMethod(string httpMethod)
```

Comment:
There is a very similar or maybe identical method `SqlTestHelper.ConvertRestMethodToHttpMethod`. I think we should be able to reuse that.

Reply:
Good catch. Have re-used the method